### PR TITLE
Add a format option for ROS 2 types [19866]

### PR DESCRIPTION
**Pull request by irenebm**
Merge after:
* https://github.com/eProsima/dev-utils/pull/83
* https://github.com/eProsima/DDS-Pipe/pull/69

To do:
- [x] Documentation
- [x] Tests

### CODE REVIEW

File: fastddsspy_participants/src/cpp/visualization/ModelParser.cpp [hunk 1]
```diff
@@ -70,7 +73,8 @@ void add_endpoint_to_vector(
         already_endpoints_index[endpoint.second.topic.m_topic_name] = endpoints.size();
         endpoints.push_back({
                         endpoint.second.topic.m_topic_name,
```
Comment (jepemi) [inline fastddsspy_participants/src/cpp/visualization/ModelParser.cpp:75]: Should also demangle ROS2 topic name

**Review by jepemi — CHANGES_REQUESTED** [review]

File: fastddsspy_tool/test/application/dds/AdvancedConfigurationExample/AdvancedConfigurationPublisher.cpp [hunk 2]
```diff
@@ -226,9 +235,6 @@ bool HelloWorldPublisher::init(
         wqos.ownership_strength().value = ownership_strength;
     }
```
[the remark this answers is not in the document]
Comment (irenebm) [reply fastddsspy_tool/test/application/dds/AdvancedConfigurationExample/AdvancedConfigurationPublisher.cpp:237]: by mistake, sorry

File: fastddsspy_participants/src/cpp/visualization/ModelParser.cpp (hunk 1 above)
Comment (irenebm) [reply fastddsspy_participants/src/cpp/visualization/ModelParser.cpp:75]: Agree

File: docs/rst/user_manual/configuration.rst [hunk 3]
```diff
@@ -158,6 +158,13 @@ This list takes precedence over the ``allowlist``.
 If a topic matches an expression both in the ``allowlist`` and in the ``blocklist``, the ``blocklist`` takes precedence,
 causing the data under this topic to be discarded.
 
+Topic type format
+-----------------
+
+The optional ``ros2-types`` tag enables specification of the format for storing schemas.
```
Comment (jepemi) [inline docs/rst/user_manual/configuration.rst:164]: Not storing but displaying

File: fastddsspy_tool/test/application/dds/AdvancedConfigurationExample/arg_configuration.h [hunk 4]
```diff
@@ -229,6 +230,8 @@ const option::Descriptor usage[] = {
       "  -p <str> \t--partitions=<str> \tPartitions to match separated by ';'."
       " Single or double quotes required with multiple partitions."
       " With empty string ('') no partitions used. (Default: '')." },
+    { PROFILE, 0, "x", "xml-profile",         Arg::String,
+      "  -x <str> \t--xml-profile=<str> \tProfile name to configure DomainParticipant."},
```
Comment (jepemi) [inline fastddsspy_tool/test/application/dds/AdvancedConfigurationExample/arg_configuration.h:234]: Extra space missing.
```suggestion
      "  -x <str> \t--xml-profile=<str>  \tProfile name to configure DomainParticipant."},
```

**Review by jepemi — CHANGES_REQUESTED** [review]

File: fastddsspy_tool/test/application/dds/AdvancedConfigurationExample/arg_configuration.h (hunk 4 above)
Comment (irenebm) [reply fastddsspy_tool/test/application/dds/AdvancedConfigurationExample/arg_configuration.h:234]: Not needed

**Review by jepemi — APPROVED** [review]
LGTM

**Pull request merged**